src/include/storage/index/bplustree_index.h

```diff
@@ -0,0 +1,139 @@
+#pragma once
+
+#include <functional>
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "common/managed_pointer.h"
+#include "storage/index/bplustree.h"
+#include "storage/index/index.h"
+#include "storage/index/index_defs.h"
+
+namespace noisepage::storage::index {
+template <uint8_t KeySize>
+class CompactIntsKey;
+template <uint16_t KeySize>
+class GenericKey;
+
+/**
+ * Wrapper around B+ Tree.
+ * @tparam KeyType the type of keys stored in the B+ Tree
+ */
+template <typename KeyType>
+class BPlusTreeIndex final : public Index {
+  friend class IndexBuilder;
+
+ private:
+  explicit BPlusTreeIndex(IndexMetadata metadata);
+
+  const std::unique_ptr<BPlusTree<KeyType, TupleSlot>> bplustree_;
+  mutable common::SpinLatch transaction_context_latch_;  // latch used to protect transaction context
+
+ public:
+  /**
+   * @return type of the index. Note that this is the physical type, not extracted from the underlying schema or other
+   * catalog metadata. This is mostly used for debugging purposes.
+   */
+  IndexType Type() const final { return IndexType::BPLUSTREE; }
+
+  /**
+   * Invoke garbage collection on the index.
+   */
+  void PerformGarbageCollection() final;
```

Comment:
This function is implemented in the generic index wrapper and a change here will have a bigger blast radius. This will need changes to the corresponding implementations in BwTree and Hash index. I'll keep this in mind and make the change in the next PR removing the BwTree.